src/ripple/rpc/tests/KeyGeneration.test.cpp

```diff
@@ -164,10 +197,50 @@ class KeypairForSignature_test : public ripple::TestSuite
         testLegacySecret (secp256k1_strings.master_seed_hex);
     }
 
+    void testInvalidKeyType (char const* keyType)
+    {
+        testcase (keyType);
+
+        Json::Value params;
+        params[jss::key_type] = keyType;
+        params[jss::passphrase] = common::passphrase;
+
+        Json::Value error;
+        (void) keypairForSignature (params, error);
```

Comment:
Its pretty clear that modification of the `error` parameter is a side-effect, using any of the common definitions of side effect:
http://en.wikipedia.org/wiki/Side_effect_%28computer_science%29

This is consistent with my original interpretation of the function: `keypairForSignature` has a side effect, it modifies one of its parameters, and `(void)` is extraneous.